README.md
# EvolvedIRC
A Mirror of the old EvolvedIRC Project of yesteryear

This repo contains the last checkin of all EvolvedIRC source from CVS.

The code should still compile under Visual Basic 6


changelog.txt
##########################
## EvolvedIRC ChangeLog ##
##########################
#
# DJ = DJ_Dark   (djdark@gmail.com)
#
# $Id: changelog.txt,v 1.6 2005/06/30 22:56:55 dj_dark Exp $
#
#####June 2004#####
#
[JUN-10/11] - DJ - Alpha Version 0.0.1
[JUN-10/11] - DJ - Working on getting all the IRC Client commands in and making a better UI
[JUN-11] - DJ - Added About Dialog and in frmMain/IRC added a Menu bar thats all for now
[JUN-11] - DJ - Tryed adding MODE command but with no luck so it's has been commented out for now unless someone wants to give it a shoot
[JUN-11] - DJ - Added Manifest file WARNING!: Only use the manifest file if your running Windows XP, 2003 or Longhorn Alpha
[JUN-12] - DJ - Got /mode command working syntax /mode #channel +/-Mode and/or Nick
[JUN-12] - DJ - Just change the Nick to EvolvedIRC|Test and the User to EvolvedIRC
[JUN-12] - DJ - Fixed a little problem with the RichTextBox not being able to scroll and copy.
[JUN-12] - DJ - Fixed a little problem with the RichTextBox not being able to scroll and copy.

#####September 2004#####
#
[SEP-2] - DJ - EvolvedIRC Win32 is back and still coded in VB6 and it's got a code name now, Grasshopper, this is Build 0001, JOIN work, ME works NICK and NICKSERV works, NICKSERV is just PRIVMSG nickserv, so no need to use /msg NICKSERV now ^_^ but THIS BUILE WILL ONLY CONNECT TO FREENODE.NET!
[SEP-7] - DJ - Started work on INI read/writer and trying to get all the IRC Commands in by the end of this month. Tomorrow is My Birthday ^_^(This is random)
[SEP-7] - DJ - started to put the commands in a Module and removed the mod_Config so I could compile te new binary.

#####October 2004#####
#
[OCT-21] - DJ - Commit of EvolvedIRC 0.1.3 Build 0003. Alot has been updated in this build.
[OCT-21] - DJ - 1). You can now change your nick and the client remembers it.
[OCT-21] - DJ - 2). You can now join any channel and chat freely now unlike the older builds where you were stuck chatting in #lobby all the time.
[OCT-21] - DJ - 3). More IRC Commands.
[OCT-21] - DJ - 4). Still no support for connecting to IRCX servers, we are working on this.
[OCT-21] - DJ - 5). New options dialog, will be worked on as much as possable.
[OCT-21] - DJ - 6). Still no formatting done on incoming text and you still can't see your out going text.
[OCT-21] - DJ - Will I think thats about it, besides having a new developer on the team.


#####January 2005#####
#
[JAN-03] - DJ - Commit of all the updates since OCT 21.
[JAN-28] - DJ - fixxing CVS(at least trying to anyway).
[Jan-28] - DJ - Added new config file to CVS (options.xml). Fixxed CVS(I hope -_-). There is a minor bug in EvolvedIRC witch keeps it from connecting to normal IRC servers, I still haven't tracked down the problem as of yet. This may be the last comit you will see in VB6 code as the EvolvedIRC team will be converting EvolvedIRC to C/C++ ^_^ (Yes it _will_ run on Mac OS(X), Windows, Linux and Unix).

#####February 2005#####
#
[Feb-1] - DJ - Updated changelog.txt
[Feb-11] - DJ - Changed the XML config file.

#####March 2005#####
#
[Mar-1] - DJ - Fixxed bug #1117389 but still will not connect to Dancer IRCd.
[Mar-2] - DJ - Got EvolvedIRC to connect to dancer IRCd.

#####June 2005#####
#
[Jun-30] - DJ - The Finally commit of EvolvedIRC VB, we are moving the project to C/C++ and we're trying to make it Cross-Platform compatible :). Cya When we start EvolvedIRC C =D.

modules/modEIRC.bas
Attribute VB_Name = "modEIRC"
'EvolvedIRC is (C)  Matthew Sporich.
'----------------------------------------------------------
'You must include this notice in any modifications you make. You must additionally
'follow the GPL's provisions for sourcecode distribution and binary distribution.
'If you are not familiar with the GPL, please read LICENSE.TXT.
'(you are welcome to add a "Based On" line above this notice, but this notice must
'remain intact!)
'Released under the GNU General Public License
'Contact information: Matthew Sporich (DJ_Dark) <djdark@gmail.com>
'                     Website <http://quantump.net/>
'
' $Id: modMain.bas,v 1.3 2005/03/02 23:47:25 dj_dark Exp $
'
'
'This program is free software.
'You can redistribute it and/or modify it under the terms of the
'GNU General Public License as published by the Free Software Foundation; either version 2 of the License,
'or (at your option) any later version.
'
'This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY.
'Without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
'See the GNU General Public License for more details.
'
'You should have received a copy of the GNU General Public License along with this program.
'if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA

Option Explicit

'`mIRC format' Constants
'These characters are used by mIRC.
'CTCP: To indicate a CTCP request or reply the hole message is included in two of these characters
'Sample Request:
'VERSION
Public Const MIRC_CTCP As String = "" 'CTCP Request or Reply = ChrW$(1)
'For a message to appear in bold, it is included in the bold characters:
'This is some bold text
Public Const MIRC_BOLD As String = "" 'Bold = ChrW$(2)
'Some colored text is included in two color characters, the first followed by one or two numbers seperated by comma:
'1,2This is some colored text
'The first number is the forecolor whereas the second is the background color.
'The numbers are between 0 and 15.
Public Const MIRC_COLOR As String = "" 'Color = ChrW$(3)
'For a message to appear underlined, it is included in the underline characters:
'This is some underlined text
Public Const MIRC_UNDERLINE As String = "" 'Underline = ChrW$(31)
Public Const MIRC_ITALIC As String = ""


modules/modConfig.bas
Attribute VB_Name = "modConfig"
'EvolvedIRC is (C)  Matthew Sporich.
'----------------------------------------------------------
'You must include this notice in any modifications you make. You must additionally
'follow the GPL's provisions for sourcecode distribution and binary distribution.
'If you are not familiar with the GPL, please read LICENSE.TXT.
'(you are welcome to add a "Based On" line above this notice, but this notice must
'remain intact!)
'Released under the GNU General Public License
'Contact information: Matthew Sporich (DJ_Dark) <djdark@gmail.com>
'                     Website <http://quantump.net/>
'
' $Id: modConfig.bas,v 1.4 2005/06/30 22:57:01 dj_dark Exp $
'
'
'This program is free software.
'You can redistribute it and/or modify it under the terms of the
'GNU General Public License as published by the Free Software Foundation; either version 2 of the License,
'or (at your option) any later version.
'
'This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY.
'Without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
'See the GNU General Public License for more details.
'
'You should have received a copy of the GNU General Public License along with this program.
'if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA

Option Explicit
'Public NName As String
'Public UName As String
'Public
'Public Server_AuthNotices As Boolean
'Public Server_AuthNotices As Boolean
'Public Server_AuthNotices As Boolean

Public Declare Function GetPrivateProfileString Lib "kernel32" Alias "GetPrivateProfileStringA" _
    (ByVal lpApplicationName As String, _
    ByVal lpKeyName As Any, _
    ByVal lpDefault As String, _
    ByVal lpReturnedString As String, _
    ByVal nSize As Long, _
    ByVal lpFileName As String) As Long
    
Public Declare Function WritePrivateProfileString Lib "kernel32" Alias "WritePrivateProfileStringA" _
    (ByVal lpApplicationName As String, _
    ByVal lpKeyName As Any, _
    ByVal lpString As Any, _
    ByVal lpFileName As String) As Long
    
Declare Function ShellExecute Lib "shell32.dll" Alias "ShellExecuteA" _
    (ByVal hWnd As Long, _
    ByVal lpOperation As String, _
    ByVal lpFile As String, _
    ByVal lpParameters As String, _
    ByVal lpDirectory As String, _
    ByVal nShowCmd As Long) As Long
    
Public Function ReadINI(SectionHeader As String, _
    VariableName As String, _
    FileName As String) As String
    
    Dim strReturn As String
    strReturn = String(255, Chr(0))
    ReadINI = Left$(strReturn, GetPrivateProfileString(SectionHeader, _
        ByVal VariableName, "", strReturn, Len(strReturn), FileName))
End Function

Public Function WriteINI(SectionHeader As String, _
    VariableName As String, _
    Value As String, _
    FileName As String)

    WriteINI = WritePrivateProfileString(SectionHeader, _
        VariableName, _
        Value, _
        FileName)
    End Function



modules/modMain.bas
Attribute VB_Name = "modMain"
'EvolvedIRC is (C)  Matthew Sporich.
'----------------------------------------------------------
'You must include this notice in any modifications you make. You must additionally
'follow the GPL's provisions for sourcecode distribution and binary distribution.
'If you are not familiar with the GPL, please read LICENSE.TXT.
'(you are welcome to add a "Based On" line above this notice, but this notice must
'remain intact!)
'Released under the GNU General Public License
'Contact information: Matthew Sporich (DJ_Dark) <djdark@gmail.com>
'                     Website <http://quantump.net/>
'
' $Id: modMain.bas,v 1.4 2005/06/30 22:57:01 dj_dark Exp $
'
'
'This program is free software.
'You can redistribute it and/or modify it under the terms of the
'GNU General Public License as published by the Free Software Foundation; either version 2 of the License,
'or (at your option) any later version.
'
'This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY.
'Without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
'See the GNU General Public License for more details.
'
'You should have received a copy of the GNU General Public License along with this program.
'if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA

'Private Declare Function SetWindowLong Lib "user32" Alias "SetWindowLongA" (ByVal hWnd As Long, ByVal nIndex As Long, ByVal dwNewLong As Long) As Long
'Private Declare Function CallWindowProc Lib "user32" Alias "CallWindowProcA" (ByVal lpPrevWndFunc As Long, ByVal hWnd As Long, ByVal msg As Long, ByVal wParam As Long, ByVal lParam As Long) As Long
Private Declare Function SendMessage Lib "user32" Alias "SendMessageA" (ByVal hWnd As Long, ByVal wMsg As Long, ByVal wParam As Long, lParam As Any) As Long
'Private Declare Sub CopyMemory Lib "kernel32" Alias "RtlMoveMemory" (Destination As Any, Source As Any, ByVal Length As Long)
'Private Declare Function ShellExecute Lib "shell32" Alias "ShellExecuteA" (ByVal hWnd As Long, ByVal lpOperation As String, ByVal lpFile As String, ByVal lpParameters As String, ByVal lpDirectory As String, ByVal nShowCmd As Long) As Long

Public Server_AuthNotices As Boolean
'Public link As SHDocVw.WebBrowser

'some constants
Public Const BuildID As String = "20050630-cvs" 'YYYYMMDD-

'indent constant
'vbTab is too huge, but somehow you can set tab lengths in the Rich Text box
'Public Const ilIndent As String = "   "
Public Const ilIndent As String = ""

'sendmessage API
'Public Declare Function SendMessage Lib "user32" Alias "SendMessageA" (ByVal hWnd As Long, ByVal wMsg As Long, ByVal wParam As Long, lParam As Any) As Long
'cursor stuff
Public Declare Function GetCursorPos Lib "user32" (lpPoint As POINTL) As Long
Public Declare Function ScreenToClient Lib "user32" (ByVal hWnd As Long, lpPoint As POINTL) As Long
Public Declare Function ClientToScreen Lib "user32" (ByVal hWnd As Long, lpPoint As POINTL) As Long

'Windows Messages Constants
Public Const WM_SETFOCUS = &H7
Public Const WM_MOUSEACTIVATE = &H21
Public Const WM_SETFONT = &H30
Public Const WM_NOTIFY = &H4E
Public Const WM_KEYDOWN = &H100
Public Const WM_KEYUP = &H101
Public Const WM_CHAR = &H102
Public Const WM_COMMAND = &H111
Public Const WM_HSCROLL = &H114
Public Const WM_VSCROLL = &H115
Public Const WM_MOUSEFIRST = &H200
Public Const WM_MOUSEMOVE = &H200
Public Const WM_LBUTTONDOWN = &H201
Public Const WM_LBUTTONUP = &H202
Public Const WM_LBUTTONDBLCLK = &H203
Public Const WM_RBUTTONDOWN = &H204
Public Const WM_RBUTTONUP = &H205
Public Const WM_RBUTTONDBLCLK = &H206
Public Const WM_MBUTTONDOWN = &H207
Public Const WM_MBUTTONUP = &H208
Public Const WM_MBUTTONDBLCLK = &H208
Public Const WM_MOUSELAST = &H209
Public Const WM_USER = &H400

' Notification masks
Public Const ENM_NONE = &H0
Public Const ENM_CHANGE = &H1
Public Const ENM_UPDATE = &H2
Public Const ENM_SCROLL = &H4
Public Const ENM_SCROLLEVENTS = &H8
Public Const ENM_DRAGDROPDONE = &H10
Public Const ENM_PARAGRAPHEXPANDED = &H20
Public Const ENM_KEYEVENTS = &H10000
Public Const ENM_MOUSEEVENTS = &H20000
Public Const ENM_REQUESTRESIZE = &H40000
Public Const ENM_SELCHANGE = &H80000
Public Const ENM_DROPFILES = &H100000
Public Const ENM_PROTECTED = &H200000
Public Const ENM_CORRECTTEXT = &H400000
Public Const ENM_LANGCHANGE = &H1000000
Public Const ENM_OBJECTPOSITIONS = &H2000000
Public Const ENM_LINK = &H4000000

' RichEdit messages
Public Const EM_GETRECT = &HB2
Public Const EM_SETRECT = &HB3
Public Const EM_GETMODIFY = &HB8&
Public Const EM_SETMODIFY = &HB9&
Public Const EM_GETLINECOUNT = &HBA&
Public Const EM_LINEINDEX = &HBB&
Public Const EM_CANUNDO = &HC6&
Public Const EM_EMPTYUNDOBUFFER = &HCD&
Public Const EM_GETLIMITTEXT = WM_USER + 37
Public Const EM_CANPASTE = WM_USER + 50
Public Const EM_EXLIMITTEXT = WM_USER + 53
Public Const EM_EXLINEFROMCHAR = WM_USER + 54
Public Const EM_FORMATRANGE = WM_USER + 57
Public Const EM_GETCHARFORMAT = WM_USER + 58
Public Const EM_GETOLEINTERFACE = WM_USER + 60
Public Const EM_SETBKGNDCOLOR = WM_USER + 67
Public Const EM_SETCHARFORMAT = WM_USER + 68
Public Const EM_SETEVENTMASK = WM_USER + 69
Public Const EM_SETOLECALLBACK = WM_USER + 70
Public Const EM_SETTARGETDEVICE = WM_USER + 72
Public Const EM_SETUNDOLIMIT = WM_USER + 82
Public Const EM_CANREDO = WM_USER + 85
Public Const EM_GETUNDONAME = WM_USER + 86
Public Const EM_GETREDONAME = WM_USER + 87
Public Const EM_AUTOURLDETECT = WM_USER + 91
Public Const EM_GETAUTOURLDETECT = WM_USER + 92
Public Const EM_SETEDITSTYLE = WM_USER + 204
Public Const EM_GETEDITSTYLE = WM_USER + 205
Public Const EM_OUTLINE = WM_USER + 220
Public Const EM_GETZOOM = WM_USER + 224
Public Const EM_SETZOOM = WM_USER + 225

Public Const EN_LINK = &H70B&

'some types
Type NMHDR
   hwndFrom As Long
   idFrom As Long
   code As Long
End Type

Type CHARRANGE
   cpMin As Long
   cpMax As Long
End Type

Type ENLINK
   NMHDR As NMHDR
   msg As Integer
   wParam As Long
   lParam As Long
   chrg As CHARRANGE
End Type

Type MSGFILTER
   NMHDR As NMHDR
   msg As Long
   wParam As Long
   lParam As Long
End Type

Type POINTL
    x As Long
    y As Long
End Type

Public Type RECT
   Left As Long
   Top As Long
   Right As Long
   Bottom As Long
End Type

Public Enum ilBulletTypes
  Arrow = 1
  Dot = 2
  Info = 3
End Enum

Sub Main()
On Error Resume Next
Load frmMSplash
frmSplash.Show
End Sub

Public Sub RTF_SetColor(lngColor As Long)
On Error Resume Next
frmMain.rtfBuffer.SelStart = Len(frmMain.rtfBuffer.Text)
frmMain.rtfBuffer.SelColor = lngColor
End Sub
Public Sub RTF_SetBold(blnBold As Boolean)
On Error Resume Next
frmMain.rtfBuffer.SelStart = Len(frmMain.rtfBuffer.Text)
frmMain.rtfBuffer.SelBold = blnBold
End Sub
Public Sub RTF_AddText(strText As String)
On Error Resume Next
frmMain.rtfBuffer.SelStart = Len(frmMain.rtfBuffer.Text)
frmMain.rtfBuffer.SelText = strText
End Sub
Public Sub InternalDebug(strText As String)
Dim F As Long
F = FreeFile
Open App.Path & "\debug.log" For Append As F
Print #F, "[" & Now & "] " & strText & CrLf
Close #F
Debug.Print strText
End Sub
Public Sub RTF_AutoURLDetect(blnAutoURLDetect As Boolean)
SendMessage frmMain.rtfBuffer.hWnd, EM_AUTOURLDETECT, Abs(blnAutoURLDetect), ByVal 0&
End Sub
Public Sub RTF_AddBullet(BulletType As ilBulletTypes, lngColor As Long)
If BulletType = Dot Then
  frmMain.rtfBuffer.SelFontName = "Wingdings"
  RTF_SetColor lngColor
  frmMain.rtfBuffer.SelStart = Len(frmMain.rtfBuffer.Text)
  RTF_AddText "w"
  frmMain.rtfBuffer.SelFontName = "Tahoma"
ElseIf BulletType = Arrow Then
  frmMain.rtfBuffer.SelFontName = "Webdings"
  RTF_SetColor lngColor
  RTF_AddText "4"
  frmMain.rtfBuffer.SelFontName = "Tahoma"
End If
End Sub

Public Sub RTF_SetViewRect(Optional Left, Optional Top, Optional Right, Optional Bottom)
Dim R As RECT
   ' Get the current rectangle
   RTF_GetViewRect R.Left, R.Top, R.Right, R.Bottom

   ' Set the new values
   If Not IsMissing(Left) Then R.Left = CLng(Left)
   If Not IsMissing(Top) Then R.Top = CLng(Top)
   If Not IsMissing(Right) Then R.Right = CLng(Right)
   If Not IsMissing(Bottom) Then R.Bottom = CLng(Bottom)

   ' Set the new rectangle
   SendMessage frmMain.rtfBuffer.hWnd, EM_SETRECT, 0, R
End Sub

Public Sub RTF_GetViewRect(Optional Left As Long, Optional Top As Long, Optional Right As Long, Optional Bottom As Long)
Dim tRECT As RECT
   ' Get the rectangle
   SendMessage frmMain.rtfBuffer.hWnd, EM_GETRECT, 0, tRECT

   With tRECT
      Left = .Left
      Top = .Top
      Right = .Right
      Bottom = .Bottom
      
      Debug.Print "Current ViewRect: L=" & .Left & " T=" & .Top & " R=" & .Right & " B=" & .Bottom
   End With
End Sub

Public Sub RTF_Indent()
frmMain.rtfBuffer.SelStart = Len(frmMain.rtfBuffer.Text)
frmMain.rtfBuffer.SelText = "   "
frmMain.rtfBuffer.SelStart = Len(frmMain.rtfBuffer.Text)
End Sub


